rclcpp/src/rclcpp/node_interfaces/node_graph.cpp

```diff
@@ -79,28 +79,30 @@ NodeGraph::get_topic_names_and_types() const
 std::vector<std::string>
 NodeGraph::get_node_names() const
 {
-  rcl_string_array_t node_names_c =
-    rcl_get_zero_initialized_string_array();
+  utilities_string_array_t node_names_c =
+    utilities_get_zero_initialized_string_array();
 
   auto ret = rcl_get_node_names(node_base_->get_rcl_node_handle(),
       &node_names_c);
-  if (ret != RMW_RET_OK) {
-    if (rmw_destroy_node_names(&node_names_c) != RMW_RET_OK) {
+  if (ret != RCL_RET_OK) {
+    if (utilities_string_array_fini(&node_names_c) != UTILITIES_RET_OK) {
       RMW_SET_ERROR_MSG("Fatal: Leaking node_name memory.");
```

Comment:
Sorry, I know this isn't part of this pr, but this looks wrong. This should either print to stderr or throw an exception, probably the latter. No one is going to check the rmw error state after this function is called.